core/src/main/java/com/linecorp/armeria/client/endpoint/WeightedRoundRobinStrategy.java

```diff
@@ -53,16 +53,22 @@ public EndpointSelector newSelector(EndpointGroup endpointGroup) {
     private static final class WeightedRoundRobinSelector extends AbstractEndpointSelector {
 
         private final AtomicInteger sequence = new AtomicInteger();
+        @Nullable
         private volatile EndpointsAndWeights endpointsAndWeights;
 
         WeightedRoundRobinSelector(EndpointGroup endpointGroup) {
             super(endpointGroup);
-            endpointsAndWeights = new EndpointsAndWeights(endpointGroup.endpoints());
-            endpointGroup.addListener(endpoints -> endpointsAndWeights = new EndpointsAndWeights(endpoints));
+            endpointGroup.addListener(endpoints -> endpointsAndWeights = new EndpointsAndWeights(endpoints),
+                                      true);
         }
 
         @Override
         public Endpoint selectNow(ClientRequestContext ctx) {
+            final EndpointsAndWeights endpointsAndWeights = this.endpointsAndWeights;
+            if (endpointsAndWeights == null) {
```

Comment:
We probably don't need this because `endpointsAndWeights` is always initialized?

Reply:
If an endpoint is not initialized, the listener will be called until `whenReady()` complete.

Reply:
Ah ha, that's correct. 😅